src/Tasks/XamlTaskFactory/TaskParser.cs

```diff
@@ -307,8 +307,13 @@ private bool ParseParameter(XamlTypes.BaseProperty baseProperty, LinkedList<Prop
             // generate the list of parameters in order
             if (!_switchesAdded.Contains(propertyToAdd.Name))
             {
+                _switchesAdded.Add(propertyToAdd.Name);
```

Comment:
It looks like `_switchesAdded` wasn't used previously, and `_switchOrderList` was used only as an enumerable. Since Contains is noticeably faster for a set than for a list, perhaps replace all uses of _switchOrderList with _switchesAdded?

Reply:
no, original ordering of `_switchOrderList` is used after parsing the task, line 599 of the `CommandLineGenerator` class (`foreach (string propertyName in _switchOrderList)`)

however, the opposite may be true, because, the typical property count of a task is in the range of ~1..10, and linear search outperforms hashset for very small item counts and task parsing is not executed very often during a build, so we could eliminate the _switchesAdded variable if we would really want to eliminate one of them

Reply:
I saw that, but I didn't think order really mattered for that loop—command lines shouldn't care about the order in which switches are presented—but upon reflection, that might break someone's tests.

I commented to @BenVillalobos that I'd like it better as a TreeSet, which would beat both versions on (almost) any normal-sized input as far as time. (Also assuming nodes aren't too far apart in memory.) I would be slightly worried by the "typical" part without asking Rainer's opinion, but I think I'd be content with how you originally had it or how it is now. Approved, pending @rainersigwald's approval.

Reply:
I agree that the order of options can be _usually_ discarded, but there are many edge cases where the order becomes important: for example, CL documentation says: "If CL encounters conflicting options, it uses the rightmost option." (https://docs.microsoft.com/en-us/cpp/build/reference/order-of-cl-options?view=vs-2019), also it is general practice in the unix world that everything after a `--` won't be handled as an option but a filename, so if you would reorder that then you would get a lot of file not found errors when the program tries to load the switch as a file

Reply:
I agree that we should preserve order, and that the N here should be low enough that the lookup being N^2 shouldn't be a big problem.

But let's double check that against VS perf tests. @Forgind can you push this to an exp/ branch and see if RPS is green?